### PR TITLE
pmenu: init at 3.0.1

### PR DESCRIPTION
**Pull request by azahi**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/pmenu/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  version = "3.0.1";
+
+  src = fetchFromGitHub {
+    owner = "phillbush";
+    repo = "pmenu";
+    rev = "v${version}";
+    sha256 = "sha256-xeOiJEOPz5QEMlWP6bWhTjmj4tfNqh3rsEVmnKvrKuM=";
+  };
+
+  buildInputs = with xorg; [
```
Comment (AndersonTorres) [inline pkgs/tools/X11/pmenu/default.nix:21]: X11 dependencies should be declared explicitly in the input set, not as subsets of xorg.

File: pkgs/tools/X11/pmenu/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, writeText
```
Comment (AndersonTorres) [inline pkgs/tools/X11/pmenu/default.nix:4]: This input is not being used.

File: pkgs/tools/X11/pmenu/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    fontconfig
+    imlib2
+    libX11
+    libXext
+    libXft
+    libXinerama
+    libXrender
+  ];
+
+  prePatch = ''
```
Comment (AndersonTorres) [inline pkgs/tools/X11/pmenu/default.nix:31]: It should be a `postPatch`.

`pre<??>` phases are useful to prepare the `<??>` phase before it runs, whereas `post<??>` serve as a "fixup" for any thing the `<??>` phase left out.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by azahi** [comment]
I've modified inputs a bit. Now there's a `conf` argument to override the default configuration.

**Pull request merged**